lib/kubeclient/common.rb

```diff
@@ -73,8 +73,8 @@ def self.define_entity_methods(entity_types)
         entity_name_plural = pluralize_entity(entity_name)
 
         # get all entities of a type e.g. get_nodes, get_pods, etc.
-        define_method("get_#{entity_name_plural}") do |options = {}|
-          get_entities(entity_type, klass, options)
+        define_method("get_#{entity_name_plural}") do |namespace:nil, options:{}|
+          get_entities(entity_type, klass, namespace, options)
```

Comment:
> Couldn't we just include namespace in the options hash? It feels weird to say:
> 
> client.get_events(namespace: 'default', options: { label_selector: 'foo=bar' })
> rather than:
> client.get_events(namespace: 'default', label_selector: 'foo=bar')

yep, based on your and @iterion feedback, I revised it to be like it.